### PR TITLE
add unit test for platform_data

### PR DESCRIPTION
**Pull request by aranguyen**

### CODE REVIEW

**Comment by aranguyen** [comment]
@katre the .bazelci/presubmit.yml is kept minimal for now so I can set up the ci pipeline. Per guidance for setting up the pipeline, we can iterate on this https://github.com/bazelbuild/continuous-integration/issues/1852

File: .bazelci/presubmit.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+build_targets: &build_targets
```
Comment (katre) [inline .bazelci/presubmit.yml]: Get rid of these and just define them inline in the `tasks` section.

File: platform_data/test/analysis_tests.bzl [hunk 2]
```diff
@@ -0,0 +1,71 @@
+"""TODO(aranguyen): Write module docstring."""
```
Comment (katre) [inline platform_data/test/analysis_tests.bzl]: Go ahead and do this, please.

**Review by katre — CHANGES_REQUESTED** [review]

File: WORKSPACE
Comment (katre) [inline WORKSPACE:1]: We should have a MODULE.bazel instead, and eventually publish to BCR

Comment (aranguyen) [reply WORKSPACE:1]: srry i missed this yesterday. What's BCR?

Comment (katre) [reply WORKSPACE:1]: It's the [Bazel Central Registry](https://registry.bazel.build). See the [directions for adding a module](https://github.com/bazelbuild/bazel-central-registry/blob/main/docs/README.md#contribute-a-bazel-module).

Comment (aranguyen) [reply WORKSPACE:1]: Done. Thanks!

**Comment by aranguyen** [comment]
@katre beside the test, I also added config files for BCR App. I think I need to have these files checked in before I can cut a release according to https://github.com/bazel-contrib/publish-to-bcr

File: .bcr/metadata.template.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
+{
+  "homepage": "https://github.com/aranguyen/rules_platform",
```
Comment (katre) [inline .bcr/metadata.template.json]: This should be rules_platform, not aranguyen, right?

File: .bcr/metadata.template.json [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  "maintainers": [
+    {
+      "name": "Ara Nguyen",
+      "email": "aranguyen@google.com",
+      "github": "aranguyen"
```
Comment (katre) [inline .bcr/metadata.template.json:6]: What are other bazel team projects doing? Should this be Open Source Team? blaze-configurability-team@google.com?

File: platform_data/test/BUILD
Comment (katre) [inline platform_data/test/BUILD:1]: Do all the files need the license header?

**Review by katre — CHANGES_REQUESTED** [review]
Thanks, this is looking good. Get @Wyverald to review for the BCR side.

File: .bcr/metadata.template.json (hunk 3 above)
Comment (aranguyen) [reply .bcr/metadata.template.json]: oh good catch. This should be https://github.com/bazelbuild/rules_platform. I'll update in the morning

File: platform_data/test/BUILD
Comment (aranguyen) [reply platform_data/test/BUILD:1]: I don't think so. I've check other repo (rules_java), only .bzl files need and not BUILD files. And I didn't see anymore complaints related to LICENSE once I added it for the .bzl files.

File: .bcr/metadata.template.json (hunk 4 above)
Comment (aranguyen) [reply .bcr/metadata.template.json:6]: This is what `rules_java` has https://github.com/bazelbuild/rules_java/blob/master/.bcr/metadata.template.json. I'll do the same

File: .bcr/metadata.template.json [hunk 5]
```diff
@@ -0,0 +1,9 @@
+{
+  "homepage": "https://github.com/bazelbuild/rules_platform",
+  "maintainers": [],
```
Comment (Wyverald) [inline .bcr/metadata.template.json]: please add yourself as a maintainer here

File: .bcr/presubmit.yml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+bcr_test_module:
+  module_path: "e2e/bzlmod"
+  matrix:
+    platform: ["debian10", "macos", "ubuntu2004", "windows"]
```
Comment (Wyverald) [inline .bcr/presubmit.yml:4]: you also need the bazel version here; this is a new requirement in BCR. see https://github.com/bazelbuild/bazel-central-registry/commit/3e3619726711fb2db7c9a484e8689dd115cd38d7#diff-7a0997a0795bbec7de12251566f182951f509eb0574ba7e96dd5d66800e3d68fR10 for a random example

File: MODULE.bazel [hunk 7]
```diff
@@ -0,0 +1,11 @@
+"""Bazel build and test dependencies."""
+
+# NOTE: When editing this file, also update the lockfile.
+#   bazel mod deps --lockfile_mode=update
+
+module(
+    name = "rules_platform",
+    repo_name = "rules_platform",
```
Comment (Wyverald) [inline MODULE.bazel]: the repo_name line is redundant -- it defaults to the same as `name`

File: platform_data/test/BUILD
Comment (katre) [reply platform_data/test/BUILD:1]: (Argh, I had a long comment and it was eaten when I hit "unresolve". Grr github).

This seems like an important point, so let's figure out who can give a definitive answer.

Some data points:
- [rules_android](https://github.com/bazelbuild/rules_android/blob/main/mobile_install/mi.bzl) - Most files have a license header
- [rules_rust](https://github.com/bazelbuild/rules_rust/blob/main/crate_universe/defs.bzl) - Most files don't have a license header
  - This project is community maintained
- [rules_java](https://github.com/bazelbuild/rules_java/blob/master/toolchains/jdk_build_file.bzl) - Most files have a license header
- [rules_cc](https://github.com/bazelbuild/rules_cc) - A mix! [Some files](https://github.com/bazelbuild/rules_cc/blob/main/cc/defs.bzl) have a license header, but [others](https://github.com/bazelbuild/rules_cc/blob/main/cc/repositories.bzl) do not.

File: MODULE.bazel (hunk 7 above)
Comment (aranguyen) [reply MODULE.bazel]: Removed. Thanks!

File: .bcr/presubmit.yml (hunk 6 above)
Comment (aranguyen) [reply .bcr/presubmit.yml:4]: Done. Thanks!

File: .bcr/metadata.template.json (hunk 5 above)
Comment (aranguyen) [reply .bcr/metadata.template.json]: Done. Thanks!

File: platform_data/test/analysis_tests.bzl (hunk 2 above)
Comment (aranguyen) [reply platform_data/test/analysis_tests.bzl]: Done. THanks!

**Review by Wyverald — APPROVED** [review]

File: platform_data/test/BUILD
Comment (aranguyen) [reply platform_data/test/BUILD:1]: reading internal guidance (I'll forward you the link on chat), we're supposed to add license header for all files that take the format of source file and supports file comments. So i'lll go ahead an add them

Comment (katre) [reply platform_data/test/BUILD:1]: Thanks!

File: platform_data/BUILD.bazel [hunk 8]
```diff
@@ -1,3 +1,17 @@
+# Copyright 2023 Google LLC
```
Comment (katre) [inline platform_data/BUILD.bazel]: Copyright 2024 everywhere

**Review by katre — CHANGES_REQUESTED** [review]

File: platform_data/BUILD.bazel (hunk 8 above)
Comment (aranguyen) [reply platform_data/BUILD.bazel]: Done. Thanks!

**Review by katre — APPROVED** [review]

**Pull request merged**